lib/resources/pam.rb

```diff
@@ -0,0 +1,370 @@
+      end
+
+      return retval
+    end
+    alias_method :match, :include?
+
+    # An alias for setting `:exact => true` in the `include` method
+    def include_exactly?(rules, opts={})
+      include?(rules, opts.merge({:exact => true}))
+    end
+    alias_method :match_exactly, :include_exactly?
+
+    # Convert the data structure to an Array suitable for an RSpec diff
+    #
+    # @return [Array[String]]
+    def to_a
+      self.sort_by{|l| l.type}.map{|l| l.to_s}
+    end
+
+    # Convert the data structure to a String
+    #
+    # @return [String]
+    def to_s
+      to_a.join("\n")
+    end
+
+    private
+
+    # Get the service name out of the configuration target
+    #
+    # @param [String] svc_name Optional name of the service that should be
+    #    returned
+    #
+    # @return String
+    def get_service_name(svc_name = nil)
+      return svc_name if svc_name
+
+      if !svc_name && @config_target.directory?
+        raise PamError, 'You must pass ":service_name" as an option!'
+      else
+        return @config_target.basename
+      end
+    end
+  end
+
+  # A single Rule object that has been processed
+  #
+  # Rule equality is a fuzzy match that can accept regular expression matches
+  # within the string to compare
+  class Rule
+    attr_reader :to_s
+    attr_reader :service, :silent, :type, :control, :module_path, :module_arguments
+
+    def initialize(rule, opts = {})
+      @to_s = rule.strip.gsub(/\s+/,' ')
+
+      rule_regex = <<-'EOM'
+        # Start of Rule
+          ^
+        # Ignore initial Whitespace
+          \s*
+        # Capture Silent Flag
+          (?<silent>-)?
+      EOM
+
+      unless opts[:service_name]
+        rule_regex += <<-'EOM'
+          # Capture Service
+            (?<service_name>.+?)\s+
+        EOM
+      end
+
+      rule_regex += <<-'EOM'
+        # Capture Type
+          (?<type>.+?)\s+
+        # Capture Control
+          (?<control>(\[.+\]|.+?))\s+
+        # Capture Module Path
+          (?<module_path>.+?(\.so)?)
+        # Capture Module Args
+          (\s+(?<module_args>.+?))?
+        # End of Rule
+          $
+      EOM
+
+      match_data = rule.match(Regexp.new(rule_regex, Regexp::EXTENDED))
+
+      unless match_data
+        raise PamError, "Invalid PAM configuration rule: '#{rule}'"
+      end
+
+      @service          = opts[:service_name] ? opts[:service_name] : match_data[:service_name]
+      @silent           = match_data[:silent] == '-'
+      @type             = match_data[:type]
+      @control          = match_data[:control]
+      @module_path      = match_data[:module_path]
+      @module_arguments = match_data[:module_args] ? match_data[:module_args].strip.split(/\s+/) : []
+    end
+
+    def match?(to_cmp)
+      to_cmp = Pam::Rule.new(to_cmp, {:service_name => @service}) if to_cmp.is_a?(String)
+
+      # The simple match first
+      self.class == to_cmp.class &&
+        @service.match(Regexp.new("^#{to_cmp.service}$")) &&
```

Comment:
`obj.match(Regexp.new("^#{str_expr}$")) => obj =~ /^#{str_expr}$/`. Am I missing something?

Reply:
It was easier to read in the larger logic chaining.

Reply:
I disagree. Longer is not easier to read and the extra calls means extra reasoning about what it is doing:

```ruby
obj.match(Regexp.new("^#{str_expr}$"))
obj =~ /^#{str_expr}$/
```

and logical chaining just means even more reasoning, so please be as clear and succinct as possible.